test/integration/core/node-operators-flow.integration.ts

```diff
@@ -0,0 +1,347 @@
+import { randomPubkeys, randomSignatures } from "lib/protocol/helpers/nor";
+
+import { Snapshot } from "test/suite";
+
+type NodeOperatorState = {
+  active: boolean;
+  name: string;
+  rewardAddress: string;
+  totalVettedValidators: bigint;
+  totalExitedValidators: bigint;
+  totalAddedValidators: bigint;
+  totalDepositedValidators: bigint;
+};
+
+type NodeOperatorSummary = {
+  targetLimitMode: bigint;
+  targetValidatorsCount: bigint;
+  stuckValidatorsCount: bigint;
+  refundedValidatorsCount: bigint;
+  stuckPenaltyEndTimestamp: bigint;
+  totalExitedValidators: bigint;
+  totalDepositedValidators: bigint;
+  depositableValidatorsCount: bigint;
+};
+
+function verifyNodeOperatorStateChanges(
+  nodeOperatorAfter: NodeOperatorState,
+  nodeOperatorBefore: NodeOperatorState,
+  expectedChanges: Partial<NodeOperatorState & { addedValidators: bigint }> = {},
+) {
+  expect(nodeOperatorAfter.totalExitedValidators).to.equal(nodeOperatorBefore.totalExitedValidators);
+  expect(nodeOperatorAfter.totalAddedValidators).to.equal(
+    nodeOperatorBefore.totalAddedValidators + (expectedChanges.addedValidators || 0n),
+  );
+  expect(nodeOperatorAfter.totalDepositedValidators).to.equal(nodeOperatorBefore.totalDepositedValidators);
+  expect(nodeOperatorAfter.name).to.equal(nodeOperatorBefore.name);
+  expect(nodeOperatorAfter.rewardAddress).to.equal(nodeOperatorBefore.rewardAddress);
+  expect(nodeOperatorAfter.active).to.equal(expectedChanges.active ?? nodeOperatorBefore.active);
+  if (expectedChanges.totalVettedValidators !== undefined) {
+    expect(nodeOperatorAfter.totalVettedValidators).to.equal(expectedChanges.totalVettedValidators);
+  } else {
+    expect(nodeOperatorAfter.totalVettedValidators).to.equal(nodeOperatorBefore.totalVettedValidators);
+  }
+}
+
+function verifyNodeOperatorSummaryStateChanges(
+  summaryAfter: NodeOperatorSummary,
+  summaryBefore: NodeOperatorSummary,
+  expectedChanges: Partial<NodeOperatorSummary> = {},
+) {
+  expect(summaryAfter.targetLimitMode).to.equal(summaryBefore.targetLimitMode);
+  expect(summaryAfter.targetValidatorsCount).to.equal(summaryBefore.targetValidatorsCount);
+  expect(summaryAfter.stuckValidatorsCount).to.equal(summaryBefore.stuckValidatorsCount);
+  expect(summaryAfter.refundedValidatorsCount).to.equal(summaryBefore.refundedValidatorsCount);
+  expect(summaryAfter.stuckPenaltyEndTimestamp).to.equal(summaryBefore.stuckPenaltyEndTimestamp);
+  expect(summaryAfter.totalExitedValidators).to.equal(summaryBefore.totalExitedValidators);
+  expect(summaryAfter.totalDepositedValidators).to.equal(summaryBefore.totalDepositedValidators);
+  if (expectedChanges.depositableValidatorsCount !== undefined) {
+    expect(summaryAfter.depositableValidatorsCount).to.equal(expectedChanges.depositableValidatorsCount);
+  } else {
+    expect(summaryAfter.depositableValidatorsCount).to.equal(summaryBefore.depositableValidatorsCount);
+  }
+}
+
+describe("Integration: Node operators flow", () => {
```

Comment:
Looks like only happy path, and no edge cases checks

Reply:
indeed